pkg/node/manager/manager.go

```diff
@@ -385,6 +386,22 @@ func (m *manager) nodeIdentityLabels(n nodeTypes.Node) (nodeLabels labels.Labels
 	if m.conf.RemoteNodeIdentitiesEnabled() {
 		if n.IsLocal() {
 			nodeLabels = labels.NewFrom(labels.LabelHost)
+			if option.Config.PolicyCIDRSelectsNodes {
+				for _, address := range n.IPAddresses {
+					addr, ok := ip.AddrFromIP(address.IP)
+					if ok {
+						var maskLength int = 128
+						if addr.Is4() {
+							maskLength = 32
+						}
+						prefix, err := addr.Prefix(maskLength)
```

Comment:
I think we can skip the hardcoding of bitlengths?
```suggestion
						prefix, err := addr.Prefix(addr.BitLen())
```